app/components/core/TouchBox.js

```diff
@@ -22,16 +24,93 @@ const styles = StyleSheet.create({
   },
 });
 
-export const TouchBox = ({ children, activity, onPress }) => (
-  <TouchableOpacity disabled={activity ? (activity.status === 'scheduled' && !activity.nextAccess) : false} onPress={onPress}>
-    <View style={styles.box}>
-      {children}
-    </View>
-  </TouchableOpacity>
-);
+// const withPreventDoubleClick = (WrappedComponent) => {
+//   class PreventDoubleClick extends React.PureComponent {
+//     onPress = debounce(this.debouncedOnPress, 300, { leading: true, trailing: false });
+
+//     debouncedOnPress = () => {
+//       this.props.onPress && this.props.onPress();
+//     }
+
+//     render() {
+//       return <WrappedComponent {...this.props} onPress={this.onPress} />;
+//     }
+//   }
+
+//   PreventDoubleClick.displayName = `withPreventDoubleClick(${WrappedComponent.displayName ||WrappedComponent.name})`
+//   return PreventDoubleClick;
+// };
+
+// const TouchableOpacityEx = withPreventDoubleClick(TouchableOpacity);
+
+const TouchBox = ({ children, activity, onPress, getResponseInActivity, getResponseInApplet, isActivity, isApplet }) => {
+  const [touched, setTouched] = useState(false);
+  const [appletTouched, setAppletTouched] = useState(false);
+
+  useEffect(() => {
+    if (activity) {
+      getResponseInActivity(false);
+    } else {
+      getResponseInApplet(false);
+    }
+    // setTouched(false);
+  }, []);
+
+  const handlePress = () => {
+    if (activity) {
+      getResponseInActivity(true);
+      if (!touched) {
+        onPress();
+      }
+      setTouched(true);
+    } else {
+      getResponseInApplet(true);
+      if (!appletTouched) {
+        onPress();
+      }
+      setAppletTouched(true);
+    }
+  };
+
+  useEffect(() => {
+    if (activity) {
+      setTouched(isActivity);
+    }
+  }, [isActivity]);
+
+  useEffect(() => {
+    if (!activity) {
+      setAppletTouched(isApplet);
+    }
+  }, [isApplet]);
+
+  return (
+    <TouchableOpacity disabled={(activity && activity.status === 'scheduled' && !activity.nextAccess)} onPress={handlePress}>
+      <View style={styles.box}>
+        {children}
+      </View>
+    </TouchableOpacity>
+  );
+};
 
 TouchBox.propTypes = {
   children: PropTypes.node.isRequired,
   activity: PropTypes.object.isRequired,
   onPress: PropTypes.func.isRequired,
+  getResponseInActivity: PropTypes.func.isRequired,
+  getResponseInApplet: PropTypes.func.isRequired,
+  isApplet: PropTypes.func.isRequired,
```

Comment:
Prop type should be boolean.
![image](https://user-images.githubusercontent.com/31543235/80372104-98c88880-8858-11ea-9a2a-dc5139362af8.png)